src/aiida/cmdline/commands/cmd_profile.py

```diff
@@ -148,9 +149,21 @@ def profile_show(profile):
 
 
 @verdi_profile.command('setdefault')
+@deprecated_command('setdefault is deprecated, use set-default instead', version=3)
 @arguments.PROFILE(required=True, default=None)
 def profile_setdefault(profile):
-    """Set a profile as the default one."""
+    """Set a profile as the default profile."""
```

Comment:
Maybe we add the fact that it is deprecated manually such that it shows up in the help text when we do `verdi profile --help`?
```suggestion
    """Set a profile as the default profile. (deprecated)"""
```

Reply:
But I don't think we do this for other commands, so maybe it is fine to skip this and keep it as is.

Reply:
I think I would use the same method we will use in https://github.com/aiidateam/aiida-core/pull/6433 once we decided there the way to go

Reply:
Agreed, let's wrap that one up first, then you can use that method here as well

Reply:
I now have the same pattern as in `verdi setup`